src/ModelParser/VisibilityAwarePropertyAccessGuesser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Liip\MetadataParser\ModelParser;
+
+use Liip\MetadataParser\Exception\ParseException;
+use Liip\MetadataParser\Metadata\PropertyAccessor;
+use Liip\MetadataParser\ModelParser\RawMetadata\PropertyVariationMetadata;
+use Liip\MetadataParser\ModelParser\RawMetadata\RawClassMetadata;
+use ReflectionClass;
+use ReflectionException;
+
+class VisibilityAwarePropertyAccessGuesser implements ModelParserInterface
+{
+    public function parse(RawClassMetadata $classMetadata): void
+    {
+        try {
+            $reflClass = new ReflectionClass($classMetadata->getClassName());
+        } catch (ReflectionException $e) {
+            throw ParseException::classNotFound($classMetadata->getClassName(), $e);
+        }
+
+        foreach ($classMetadata->getPropertyCollections() as $propertyCollection) {
```

Comment:
i would prefer to loop over the reflClass like we do in ReflectionParser and update the variation with the name, rather than update each variation.

Reply:
with virtual properties there can be multiple fields with the same name, but that will be on methods, not on the property itself. the property itself afaict is always unique.